app/src/main/res/layout/fragment_training.xml

```diff
@@ -152,13 +152,14 @@
                 android:layout_height="match_parent" />
 
         </LinearLayout>
+        <ProgressBar
+            android:id="@+id/progressBar"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_centerInParent="true" />
```

Comment:
This attribute won't work in constraint layout to keep the progress bar in center. You need to use constraintStartToStartOf=parent
constraintEndToEndOf=parent
constraintTopToTopOf=parent
,...

Reply:
I added those attributes you mentioned @ujjwalagrawal17, please review.

Reply:
Bottom to bottom of attribute is missing.